core/directory.go

```diff
@@ -0,0 +1,291 @@
+package core
+
+import (
+	"context"
+	"path"
+
+	"github.com/moby/buildkit/client/llb"
+	bkgw "github.com/moby/buildkit/frontend/gateway/client"
+	"github.com/moby/buildkit/solver/pb"
+	"go.dagger.io/dagger/core/schema"
+	"go.dagger.io/dagger/router"
+)
+
+// Directory is a content-addressed directory.
+type Directory struct {
```

Comment:
I'm open to rearranging things, but yeah I figured I'd start with just putting it all together.

One option could be to move the schema glue out, but I think I would still prefer one big package for all the models to live in since they're inter-dependent (possibly with circles at some point?) and share a lot of common behavior (e.g. `FooID` scalar definitions and encoding/decoding). If we were to do per-type packages I think we'd end up with a `/util` package somewhere which feels a little awkward to me.